### ISSUE TITLE
interface ContactInfo: metadata distinguishing email/tel instances? 

### ISSUE THREAD

**Issue by inexorabletash** [beverloo/contact-api#4]
An OS's contact information will have metadata distinguishing email/tel instances (e.g. "work", "home"). Should this API expose that?

(if we say "no" it should be called out explicitly)

**Comment by beverloo** [comment]
I have considered this, but I'm not sure where we would draw the line. You could make the same argument about pretty much other data types. Labels can be repeated as well, so we'd end up with something like:

```webidl
interface ContactProperty {
    readonly attribute string? label;
    readonly attribute string value;
};

interface ContactInfo {
    readonly attribute sequence<ContactProperty>? name;
    readonly attribute sequence<ContactProperty>? email;
    readonly attribute sequence<ContactProperty>? tel;
};
```

Would love to hear from developers on this :).

**Comment by annevk** [comment]
(I'd also want the opportunity to detail what I want to share from a contact. Not all email addresses might be shareable for instance.)

**Comment by tomayac** [comment]
I agree that in the end this becomes more a question of the picker UI and doesn't need to be surfaced to the website.

**Comment by tomayac** [comment]
I would also like to point out that `email` and `tel` return arrays, that is, typically are plural, as folks tend to store work and private details for their contacts [citation needed, but I do personally].

**Comment by annevk** [comment]
That's pretty clear from the IDL? Or are you suggesting the member names should therefore not be singular?

**Comment by tomayac** [comment]
I’m indeed suggesting if something returns multiple items that then the property name should also be plural. Principle of least surprise.
